### PR TITLE
chore: improve bash script for release

### LINKED ISSUE #2395: improve script for release

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #2395

### CODE REVIEW

**Comment by adityastic** [comment]
@iamareebjamal

File: scripts/prep-key.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #!/bin/sh
 set -e
 
-export DEPLOY_BRANCH=${DEPLOY_BRANCH:-development}
-export PUBLISH_BRANCH=${PUBLISH_BRANCH:-master}
```
Comment (iamareebjamal) [inline scripts/prep-key.sh]: Use this. We can override this from environment variables

File: scripts/prep-key.sh [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+[[ $TRAVIS_PULL_REQUEST =~ ^(false)$ && $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ && $TRAVIS_BRANCH =~ ^(development|master)$ ]] && DEV_OR_MASTER_BUILD=true || DEV_OR_MASTER_BUILD=false
+[[ $TRAVIS_BRANCH =~ ^(master)$ && $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ && $TRAVIS_PULL_REQUEST_SLUG =~ ^(fossasia/open-event-attendee-android)$ ]] && PR_FOR_RELEASE=true || PR_FOR_RELEASE=false
 
-if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_REPO_SLUG" != "fossasia/open-event-attendee-android" ] || ! [ "$TRAVIS_BRANCH" == "$DEPLOY_BRANCH" -o "$TRAVIS_BRANCH" == "$PUBLISH_BRANCH" ]; then
```
Comment (iamareebjamal) [inline scripts/prep-key.sh]: The previous code is more readable, modify yours like this

Comment (adityastic) [reply scripts/prep-key.sh]: great, let me try to break the conditions down

File: scripts/update-apk.sh [hunk 3]
```diff
@@ -4,10 +4,17 @@ set -e
...
 git config --global user.email "noreply+travis@fossasia.org"
 
-[[ $TRAVIS_BRANCH =~ ^(master)$ && $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ ]] && IS_PUBLISH_BRANCH=true || IS_PUBLISH_BRANCH=false
-[[ $TRAVIS_BRANCH =~ ^(development)$ && $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ ]] && IS_DEV_BRANCH=true || IS_DEV_BRANCH=false
-[[ $TRAVIS_PULL_REQUEST =~ ^(false)$ && $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ && $TRAVIS_BRANCH =~ ^(development|master)$ ]] && DEV_OR_MASTER_BUILD=true || DEV_OR_MASTER_BUILD=false
-[[ $IS_PUBLISH_BRANCH && $TRAVIS_PULL_REQUEST_SLUG =~ ^(fossasia/open-event-attendee-android)$ ]] && PR_FOR_RELEASE=true || PR_FOR_RELEASE=false
+[[ $TRAVIS_REPO_SLUG =~ ^(fossasia/open-event-attendee-android)$ ]] && IS_FOSS_SLUG=true || IS_FOSS_SLUG=false
+[[ $TRAVIS_PULL_REQUEST_SLUG =~ ^(fossasia/open-event-attendee-android)$ ]] && IS_FOSS_PR=true || IS_FOSS_PR=false
+[[ $TRAVIS_BRANCH =~ ^(development|master)$ && $IS_FOSS_SLUG ]] && BRANCH_DEPLOYORDEV=true || BRANCH_DEPLOYORDEV=false
+[[ $TRAVIS_BRANCH =~ ^(master)$ && $IS_FOSS_SLUG ]] && IS_PUBLISH_BRANCH=true || IS_PUBLISH_BRANCH=false
```
Comment (iamareebjamal) [inline scripts/update-apk.sh]: Why not `IS_PUBLISH_BRANCH = [[ $TRAVIS_BRANCH =~ ^(master)$ && $IS_FOSS_SLUG ]]`

Comment (adityastic) [reply scripts/update-apk.sh]: no, conditional assignments don't work like this in bash. This was my first thought as well, but  unfortunately.

Comment (iamareebjamal) [reply scripts/update-apk.sh]: Will see

**Comment by adityastic** [comment]
@iamareebjamal

**Comment by adityastic** [comment]
@iamareebjamal I'll look into possiblities and get back to you if I found a nice workaround

**Comment by iamareebjamal** [comment]
git release is failing after fastlane release https://travis-ci.org/fossasia/open-event-attendee-android/builds/623111500?utm_source=github_status&utm_medium=notification

@adityastic

**Comment by adityastic** [comment]
@iamareebjamal sure, just give me a moment

**Pull request merged**